File: src/cbs/schemas/account.py
"""Account schemas."""

from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel

from cbs.models.account import AccountStatus, AccountType


class AccountCreate(BaseModel):
    customer_id: int
    account_type: AccountType
    currency: str = "EUR"
    interest_rate: Decimal = Decimal("0.00")
    overdraft_limit: Decimal = Decimal("0.00")


class AccountResponse(BaseModel):
    id: int
    iban: str
    customer_id: int
    account_type: AccountType
    currency: str
    balance: Decimal
    available_balance: Decimal
    status: AccountStatus
    interest_rate: Decimal
    overdraft_limit: Decimal
    created_at: datetime

    model_config = {"from_attributes": True}


class AccountUpdate(BaseModel):
    status: AccountStatus | None = None
    interest_rate: Decimal | None = None
    overdraft_limit: Decimal | None = None

File: cbs-core/src/main/java/com/cbs/core/model/Account.java
package com.cbs.core.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Account implements Serializable {
    private static final long serialVersionUID = 1L;

    private String accountId;
    private String customerId;
    private String accountType; // CURRENT, SAVINGS, TERM, CORPORATE
    private String currency; // EUR, USD, etc.
    private BigDecimal balance;
    private BigDecimal availableBalance;
    private String status; // ACTIVE, BLOCKED, CLOSED
    private String iban;
    private String bic;
    
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private LocalDateTime createdAt;
    
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private LocalDateTime lastModified;
    
    private BigDecimal interestRate;
    private String productCode;
}
